src/test/java/com/puppycrawl/tools/checkstyle/bdd/InlineConfigParser.java

```diff
@@ -557,6 +585,69 @@ private static void handleKeyValueConfig(TestInputConfiguration.Builder testInpu
         }
     }
 
+    public static Map<String, String> getPropertiesWithDefaults(String filePath, String inputPath)
+            throws IOException {
+        final Map<String, String> propertiesMap = new HashMap<>();
+        try {
+
+            final File xmlFile = new File(filePath);
+            final DocumentBuilderFactory factory = DocumentBuilderFactory.newInstance();
+            final DocumentBuilder builder = factory.newDocumentBuilder();
+            final Document document = builder.parse(xmlFile);
+            document.getDocumentElement().normalize();
+
+            final NodeList propertyNodes = document.getElementsByTagName("property");
+
+            for (int index = 0; index < propertyNodes.getLength(); index++) {
+                final Node node = propertyNodes.item(index);
+                if (node.getNodeType() == Node.ELEMENT_NODE) {
+                    final Element propertyElement = (Element) node;
+                    final String name = propertyElement.getAttribute("name");
+                    final String defaultValue = propertyElement.getAttribute("default-value");
+                    propertiesMap.put(name, defaultValue);
+                }
+            }
+        }
+        catch (Exception ex) {
+            throw new IOException("Error parsing XML file: " + filePath
+                + "input file: " + inputPath, ex);
+        }
+        return propertiesMap;
+    }
+
+    public static String modifyPath(String filePath, String replacement) {
+        final String[] parts = filePath.split(Pattern.quote(File.separator));
+        String extension = ".xml";
+        final StringBuilder newPath = new StringBuilder(128);
+        for (int index = 0; index < parts.length - 2
+            && (index < 2 || !"checks".equals(parts[index - 2])
+            && !"filters".equals(parts[index - 1])
+            && !"filefilters".equals(parts[index - 1])); index++) {
+            newPath.append(parts[index]).append(File.separator);
+        }
+
+        newPath.append(replacement);
+        final String xmlPath = newPath.toString()
+            .replace("test", "main")
+            .replace("resources-noncompilable", "resources")
+            .replace(String.join(File.separator, "com", "puppycrawl",
+                    "tools", "checkstyle"),
+                        String.join(File.separator, "com",
+                            "puppycrawl", "tools", "checkstyle", "meta"));
+        final String[] dir = xmlPath.split(Pattern.quote(File.separator));
+        int len = dir.length - 1;
+        while (!"meta".equals(dir[len])) {
+            if ("checks".equals(dir[len]) && "meta".equals(dir[len - 1])) {
+                extension = "Check.xml";
+                break;
+            }
+            len--;
```

Comment:
```suggestion
        int length = dir.length - 1;
        while (!"meta".equals(dir[length])) {
            if ("checks".equals(dir[length]) && "meta".equals(dir[length - 1])) {
                extension = "Check.xml";
                break;
            }
            length--;
```
3 chars are not so costly that we might go for the explizit code, instead of saving 3 chars. - `gth`

Reply:
done.